GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -159,6 +161,27 @@ public bool ChangeModel(string model)
                     break;
                 default:
                     throw new ArgumentOutOfRangeException(nameof(source), source, null);
+            }
+
+            if (MagShieldAmount > 0 && type == DamageType.DAMAGE_TYPE_MAGICAL)                
+            {
+                float shieldAmount = MagShieldAmount;
+                ApplyShield(-damage, ShieldType.SHIELD_MAGICAL, false);
+                damage -= shieldAmount;
+                if (MagShieldAmount < 0)
+                {
+                    MagShieldAmount = 0;
+                }
+            }
+            if (PhyShieldAmount > 0 && type == DamageType.DAMAGE_TYPE_PHYSICAL)

```

Comment:
Repetitive code.

You can remove this repetition with a simple local variable: `shieldType` to keep what is the shield type needed. As it seems that the damage types are mutually exclusive you can even compact it more.

Ans then simply:
```
                float shieldAmount = MagShieldAmount;
                ApplyShield(-damage, ShieldType.SHIELD_MAGICAL, false);
                damage -= shieldAmount;
                if (MagShieldAmount < 0)
                {
                    MagShieldAmount = 0;
                }
```